Src/PChecker/CheckerCore/Specifications/Monitor.cs

```diff
@@ -246,6 +297,16 @@ protected void RaiseGotoStateEvent(Type state)
             CheckDanglingTransition();
             PendingTransition = new Transition(Transition.Type.Goto, state, default);
         }
+        
+        public void TryAssert(bool predicate)
```

Comment:
Do we need this function? Instead, the generated code should directly call `Assert` function that this function is calling.

Reply:
Calling assert function directly results in this error: `'Assert(bool)' is inaccessible due to its protection level`, when I try setting Assert function to public, it results in: `'StateMachine.Assert(bool)' is inaccessible due to its protection level`. Is it ok if all assert functions are set to public?

Reply:
Yes.